// @syncfusion/ej2-angular-heatmap/esm2020/src/heatmap/heatmap.component.mjs
import { __decorate } from "tslib";
import { Component, ChangeDetectionStrategy } from '@angular/core';
import { ComponentBase, ComponentMixins, setValue } from '@syncfusion/ej2-angular-base';
import { HeatMap } from '@syncfusion/ej2-heatmap';
import * as i0 from "@angular/core";
export const inputs = ['allowSelection', 'backgroundColor', 'cellSettings', 'dataSource', 'dataSourceSettings', 'enableMultiSelect', 'enablePersistence', 'enableRtl', 'height', 'legendSettings', 'locale', 'margin', 'paletteSettings', 'renderingMode', 'showTooltip', 'theme', 'titleSettings', 'tooltipSettings', 'width', 'xAxis', 'yAxis'];
export const outputs = ['cellClick', 'cellDoubleClick', 'cellRender', 'cellSelected', 'created', 'legendRender', 'load', 'loaded', 'resized', 'tooltipRender'];
export const twoWays = [''];
/**
 * Represents the Angular HeatMap component.
 * This is used to customize the properties of the heatmap in order to visualize two-dimensional data, with values represented by gradient or solid color variations.
 * ```html
 * <ejs-heatmap></ejs-heatmap>
 * ```
 */
let HeatMapComponent = class HeatMapComponent extends HeatMap {
    constructor(ngEle, srenderer, viewContainerRef, injector) {
        super();
        this.ngEle = ngEle;
        this.srenderer = srenderer;
        this.viewContainerRef = viewContainerRef;
        this.injector = injector;
        this.tags = [''];
        this.element = this.ngEle.nativeElement;
        this.injectedModules = this.injectedModules || [];
        try {
            let mod = this.injector.get('HeatMapLegend');
            if (this.injectedModules.indexOf(mod) === -1) {
                this.injectedModules.push(mod);
            }
        }
        catch { }
        try {
            let mod = this.injector.get('HeatMapTooltip');
            if (this.injectedModules.indexOf(mod) === -1) {
                this.injectedModules.push(mod);
            }
        }
        catch { }
        try {
            let mod = this.injector.get('HeatMapAdaptor');
            if (this.injectedModules.indexOf(mod) === -1) {
                this.injectedModules.push(mod);
            }
        }
        catch { }
        this.registerEvents(outputs);
        this.addTwoWay.call(this, twoWays);
        setValue('currentInstance', this, this.viewContainerRef);
        this.context = new ComponentBase();
    }
    ngOnInit() {
        this.context.ngOnInit(this);
    }
    ngAfterViewInit() {
        this.context.ngAfterViewInit(this);
    }
    ngOnDestroy() {
        this.context.ngOnDestroy(this);
    }
    ngAfterContentChecked() {
        this.context.ngAfterContentChecked(this);
    }
};
HeatMapComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: HeatMapComponent, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }, { token: i0.ViewContainerRef }, { token: i0.Injector }], target: i0.ɵɵFactoryTarget.Component });
HeatMapComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.3", type: HeatMapComponent, selector: "ejs-heatmap", inputs: { allowSelection: "allowSelection", backgroundColor: "backgroundColor", cellSettings: "cellSettings", dataSource: "dataSource", dataSourceSettings: "dataSourceSettings", enableMultiSelect: "enableMultiSelect", enablePersistence: "enablePersistence", enableRtl: "enableRtl", height: "height", legendSettings: "legendSettings", locale: "locale", margin: "margin", paletteSettings: "paletteSettings", renderingMode: "renderingMode", showTooltip: "showTooltip", theme: "theme", titleSettings: "titleSettings", tooltipSettings: "tooltipSettings", width: "width", xAxis: "xAxis", yAxis: "yAxis" }, outputs: { cellClick: "cellClick", cellDoubleClick: "cellDoubleClick", cellRender: "cellRender", cellSelected: "cellSelected", created: "created", legendRender: "legendRender", load: "load", loaded: "loaded", resized: "resized", tooltipRender: "tooltipRender" }, usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
HeatMapComponent = __decorate([
    ComponentMixins([ComponentBase])
], HeatMapComponent);
export { HeatMapComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: HeatMapComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'ejs-heatmap',
                    inputs: inputs,
                    outputs: outputs,
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    queries: {}
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.Renderer2 }, { type: i0.ViewContainerRef }, { type: i0.Injector }]; } });
